test/index.js

```diff
@@ -17,31 +18,51 @@ ProductSchema.set('toJSON', {getters: true});
 
 var Product = mongoose.model('Product', ProductSchema);
 
-describe('mongoose-uuid', function(){
+describe('mongoose-uuid', function() {
   before(function() {
-    return mongoose.connect('mongodb://localhost:27017/mongoose-uuid-test')
+    return mongoose.connect('mongodb://localhost:27017/mongoose-uuid-test', { useNewUrlParser: true })
   })
 
   after(function(cb) {
-    Product.remove({}, function() {
+    Product.deleteMany({}, function() {
       mongoose.disconnect(cb);
     });
   });
 
-  it('should cast uuid strings to binary', function(){
+  it('should cast uuid strings to binary', function() {
     var product = new Product({
       _id: '7c401d91-3852-4818-985d-7e7b79f771c3',
       name: 'Some product'
     });
+
     (product._doc._id instanceof mongoose.Types.Buffer.Binary).should.equal(true);
   });
 
-  it('should convert back to text with toObject()', function(){
+  it('should cast null to null', function() {
+    var product = new Product({
+      _id: null,
+      name: 'Some product'
+    });
+
+    (product._doc._id == null).should.equal(true);
+  });
+
+  it('should not throw when given anything other than a Buffer or String', function() {
+    var product = new Product({
+      _id: 300,
```

Comment:
I'm not sure if swallowing validation errors is a good idea?

Reply:
The error is never actually thrown. I tried to make it throw many times, but it wouldn't. I believe this has to do with how we are wrapping mongoose's Buffer. I originally wrote the test to make sure it did throw, but the expect is being caught somewhere and a UUID is still output.

What would you suggest we do instead?

Reply:
I see, ok then maybe a clearer explanation could help, but feel free to leave as is.  👍